### ISSUE TITLE
Implement System.Net.Security on Unix

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#2483]

**Comment by SidharthNabar** [comment]
#2492

**Comment by vijaykota** [comment]
Targeting NegotiateStream work in RTM hence changing milestone to RTM

**Comment by vijaykota** [comment]
Changing milestone to RC2 per WCF requirement. The missing piece in System.Net.Security is NegotiateStreamand the following work may help in implementation:
1. PAL design for NegotiateStream: https://github.com/vijaykota/corefx/commits/_feature Pending work for RC2:
  a. Scrub the code moved from TFS
  b. Replace Interop.PlaceHolder.cs with actual Interop code of libgssapi_krb5 and libheimntlm
  c. Fallback logic to use NTLM credentials if GSSAPI fails. Current logic of checking status from gss_acquire_cred is not working.
2. Interop code for libgssapi: https://github.com/rajansingh10/corefx/commits/NetFeature2 Pendng work for RC2:
  a. Testing with Windows client and Unix server (for RC2, only Unix client is needed)
  b. Testing on OSX. If the libgssapi_krb5 on OSX turns out to be Heimdal version (unlike MIT in case of Ubuntu), the APIs would need some tweaks. It would make sense then to switch to libgssapi.so on Ubuntu as well
3. Interop code for libheimntlm: second commit in https://github.com/rajansingh10/corefx/pull/2 (only the Interop\libheimntlm\* files are needed). Pending work for RC2:
  a. Shimming for libheimntlm APIs and some libcrypto APIs
  b. Testing on OSX

Pending NegotiateStream work beyond v1: see #3717 and #3718

**Comment by vijaykota** [comment]
Changing milestone To Future till the Windows implementation of NegotiateStream is available.

**Comment by vijaykota** [comment]
Windows implementation is checked-in via #5156 So targeting the xplat work for v1 RTM.

**Comment by joshfree** [comment]
tagging as blocking-release since this is a required feature

**Comment by vijaykota** [comment]
The v1 support will be complete once PR# 6905 is merged to master. There are individual issues to track specific limitations in the libraries used by xplat implementation.

**Comment by joshfree** [comment]
@shrutigarg any update?

**Comment by shrutigarg** [comment]
@joshfree PR# 6905 is done and opened a pull request to merge into master. PR #7647 
In parallel we are working on the issues too and getting them merged into the master.

**Comment by shrutigarg** [comment]
NTLM implementation is done and merged . Nothing is left now in client side implementation.

**Comment by stephentoub** [comment]
> NTLM implementation is done and merged . Nothing is left now in client side implementation.

What's left then?  Can this issue be closed?

**Comment by shrutigarg** [comment]
Client side part is done .But I am unsure if this is to track server implementation too? If no , we can close it.
